app/services/browser_support.rb

```diff
@@ -0,0 +1,10 @@
+class BrowserSupport
+  def self.supported?(user_agent)
+    matcher = BrowserslistUseragent::Match.new(browser_support_config, user_agent)
```

Comment:
Unfortunately this uses a different underlying library ([`user_agent_parser`](https://rubygems.org/gems/user_agent_parser)) from the `browser` gem, so now we'd be parsing browser from user agent twice 😕

Reply:
The format of the browserlist json looks fairly straightforward, could we add a small wrapper/translation layer to the browser gem that we use? And we could also add a spec in CI to run on `browsers.json` at build time so we'd at least know if a knew browser was added that we hadn't previously handled?

```ruby
browserlist = ["and_chr 110","and_uc 13.4","chrome 109","chrome 108","edge 109","edge 108","firefox 109","firefox 108","ios_saf 16.2","ios_saf 16.1","ios_saf 16.0","ios_saf 15.6","ios_saf 14.5-14.8","op_mini all","opera 94","safari 16.2","safari 15.6","samsung 19.0"]

browser = BrowserCache.parse(user_agent)

browserlist.any? do |browserlist_version|
  slug, version_str = browserlist_version.split(' ')
  version = version_str == 'all' ? nil : version_str.to_f
  case slug
  when 'and_uc'
  when 'chrome'
    browser.chrome?(version)
  when 'op_mini'
    browser.opera_mini?(version)
  end
end
```

Reply:
Hm, yeah, that's an interesting idea.

Reply:
Alternatively, we could just add a second caching parsing/caching layer around this `BrowserslistUseragent::Match` call, so if we're parsing things twice at least we're caching twice?

Reply:
Yeah, that seems viable as well. I worried about the cache growing out of control, but I see we already have an approach for that in `BrowserCache` with a sized LRU cache, and it looks easy enough to implement. I'll consider that as well.

My first thought was whether we could swap the `browser` gem for `user_agent_parser`, but while the latter does have some of the same functionality, it doesn't have all of what we need. I'd also wondered if there was some more portable/compatible "post-parsed" value that I could carry from one to initialize the other, but that seems unlikely as well.

Reply:
Maybe we cache the results together? So instead of returning `Browser` from the cache we return `[Browser, Boolean]` and pre-compute the "is supported" bit?

Reply:
Yeah! In retrospect, that sounds like something we might want to be caching even if we found some other way to avoid re-parsing the user agent.

Reply:
Latest commits add a caching component.